File: api/Gemini.py
"""
This document interacts with Gemini API to generate multimodal outputs.
"""

import google.generativeai as genai
from dotenv import load_dotenv
import os
from PIL import Image

load_dotenv()

api_key = os.getenv("GEMINI_API_KEY")

image_path = "figma.png"
image = Image.open(image_path)

experimental_LearnLM = "learnlm-1.5-pro-experimental"

def geminiAI(text: str, image = None) -> str:
    genai.configure()
    model = genai.GenerativeModel(experimental_LearnLM)
    if image:
        response = model.generate_content([text, image])
    else: 
        response = model.generate_content(text)

    return response.text

if __name__ == "__main__":
    print(geminiAI("Whats in the picture? Also, tell me a joke about React.", image))

File: api/gemini.js
import { GoogleGenerativeAI } from "@google/generative-ai";
import dotenv from "dotenv";
import sharp from "sharp";

// Load environment variables
dotenv.config();

// Initialize Google Generative AI client
const genAI = new GoogleGenerativeAI("process.env.GEMINI_API_KEY");
const model = genAI.getGenerativeModel({ model: "learnlm-1.5-pro-experimental" });

async function geminiAI(text, image = null) {
  try {
    if (image) {
      // Convert image to buffer and encode as base64
      const imageBuffer = await sharp(image).toBuffer();
      const mimeType = "image/png";

      // Generate content with text and image
      const response = await model.generateContent([
        text,
        {
          inlineData: {
            data: imageBuffer.toString("base64"),
            mimeType,
          },
        },
      ]);

      return response.responses?.[0]?.text || "No response text received.";
    } else {
      // Generate content with only text
      const response = await model.generateContent(text);
      return response.responses?.[0]?.text || "No response text received.";
    }
  } catch (error) {
    console.error("Error in geminiAI function:", error);
    throw error;
  }
}

// Example usage
(async () => {
  try {
    const result = await geminiAI(
      "What's in the picture? Also, tell me a joke about React.",
      "figma.png"
    );
    console.log("Generated content:", result);
  } catch (error) {
    console.error("Error during execution:", error);
  }
})();
